### ISSUE TITLE
EXISTS does not work with subqueries

### ISSUE THREAD

**Issue by alsundukov** [agershun/alasql#1904]
The EXISTS function works correctly in the main query but throws an error when used in a subquery. For example:

```
res = alasql('SELECT EXISTS(SELECT "a"), a FROM ?', [[item1, item2]]);
// returns expected value

res = alasql('SELECT * FROM (SELECT EXISTS(SELECT "a"), a FROM ?)', [[item1, item2]]);
// error: Cannot read properties of undefined (reading '0')
```

Additionally, I've noticed that the function also fails when used with SET:

`SET @ress = (SELECT EXISTS(SELECT "test"));`

**Comment by mathiasrw** [comment]
Released as part of v4.5.0

**Comment by alsundukov** [comment]
Unfortunately, [this ](https://github.com/AlaSQL/alasql/commit/aeb3b00c926dcc7e7ea7c7e15c95323261c66f5d) does not fix the problem in any way. Now exists returns a Boolean value (true/false) instead of a number (0.1), but this still doesn't work for subquery(((. 

Why wasn't the test added?
